chore(docs): Update Gatsby framework version support information (#33825)
gatsbyjs/gatsby: README.md

Wondering what we've shipped recently? Check out our [release notes](https://www.gatsbyjs.com/docs/reference/release-notes) for key highlights, performance improvements, new features, and notable bugfixes.

> Also, [see here](https://www.gatsbyjs.com/docs/reference/release-notes/gatsby-version-support/) to understand the version support plans for each version of Gatsby.

Reviewer: ```suggestion
Also, read our [documentation on version support](https://www.gatsbyjs.com/docs/reference/release-notes/gatsby-version-support/) to understand our plans for each version of Gatsby.
```

Better link text than "see here"
